hnne/projector.py

```diff
@@ -86,13 +94,32 @@ class HNNE(BaseEstimator):
     """
     def __init__(
             self,
-            dim: int = 2,
+            dim: Union[Optional[int], None] = None,
+            n_components: Union[int, None] = None,
             metric: str = 'cosine',
             radius: float = 0.4,
             ann_threshold: int = 40000,
             preliminary_embedding: str = 'pca'
     ):
-        self.dim = dim
+        if (dim is None) and (n_components is None):
+            raise ValueError('Either `dim` or `n_components` must be specified.')
+
+        if (dim is not None) and (n_components is not None):
```

Comment:
To reproduce the old behavior with minimal confusion, we could override the `n_components` with the value of `dim` if `n_components == 2`, else raise the error (as it is clear that the user set both values).